datacube/api/grid_workflow.py

```diff
@@ -0,0 +1,459 @@
+                attrs=array.coords[dim].attrs,
+                fastpath=True,
+            ),
+        )
+        for dim, indexer in zip(array.dims, indexers)
+        if isinstance(indexer, slice)
+    )
+    return xr.DataArray(data, dims=dims, coords=coords, attrs=array.attrs)
+
+
+class Tile:
+    """
+    The Tile object holds a lightweight representation of a datacube result.
+
+    It is produced by :meth:`.GridWorkflow.list_cells` or :meth:`.GridWorkflow.list_tiles`.
+
+    The Tile object can be passed to :meth:`GridWorkflow.load` to be loaded into memory as
+    an :class:`xr.Dataset`.
+
+    A portion of a tile can be created by using index notation. eg:
+
+        tile[0:1, 0:1000, 0:1000]
+
+    This can be used to load small portions of data into memory, instead of having to access
+    the entire `Tile` at once.
+    """
+
+    def __init__(self, sources: xr.DataArray, geobox):
+        """Create a Tile representing a dataset that can be loaded.
+
+        :param xr.DataArray sources: An array of non-spatial dimensions of the request, holding lists of
+            datacube.storage.DatasetSource objects.
+        :param model.GeoBox geobox: The spatial footprint of the Tile
+        """
+        self.sources: xr.DataArray = sources
+        self.geobox: GeoBox = geobox
+
+    @property
+    def dims(self) -> tuple[Hashable, ...]:
+        """Names of the dimensions, eg ``('time', 'y', 'x')``
+        :return: tuple(str)
+        """
+        return self.sources.dims + self.geobox.dimensions
+
+    @property
+    def shape(self) -> tuple[int, ...]:
+        """Lengths of each dimension, eg ``(285, 4000, 4000)``
+        :return: tuple(int)
+        """
+        return self.sources.shape + self.geobox.shape
+
+    @property
+    def product(self) -> Product:
+        """
+        :rtype: datacube.model.DatasetType
+        """
+        return self.sources.values[0][0].product
+
+    def __getitem__(self, chunk):
+        sources = _fast_slice(self.sources, chunk[: len(self.sources.shape)])
+        geobox = self.geobox[chunk[len(self.sources.shape):]]
+        return Tile(sources, geobox)
+
+    # TODO(csiro) Split on time range
+    def split(self, dim: str, step: int = 1):
+        """
+        Splits along a non-spatial dimension into Tile objects with a length of 1 or more in the `dim` dimension.
+
+        :param dim: Name of the non-spatial dimension to split
+        :param step: step size to split
+        :return: tuple(key, Tile)
+        """
+        axis = self.dims.index(dim)
+        indexer = [slice(None)] * len(self.dims)
+        size = self.sources[dim].size
+        for i in range(0, size, step):
+            indexer[axis] = slice(i, min(size, i + step))
+            yield self.sources[dim].values[i], self[tuple(indexer)]
+
+    def split_by_time(self, freq: str = "A", time_dim: str = "time", **kwargs):
+        """
+        Splits along the `time` dimension, into periods, using pandas offsets, such as:
```

Comment:
Are the letters fixed, or do they get updated frequently? If they're fixed, `freq: Literal["A", "Q", "M"] = "A"` is a less liberal type for the parameter.

Reply:
They're not fixed, they come from pandas.